src/main/java/cr0s/warpdrive/config/structures/Schematic.java

```diff
@@ -65,28 +102,142 @@ public boolean loadFromXmlElement(final Element element) throws InvalidXmlExcept
+				String propertyValue = metaParts[1].trim();
+				/*
+				IProperty<? extends Comparable<?>> property = this.block.getBlockState().getProperty(propertyKey);
+				if(property == null){
+					WarpDrive.logger.warn(String.format("Found invalid block property %s for block %s", propertyKey, state));
+				}else {
+					Optional<? extends Comparable<?>> result = property.parseValue(propertyValue);
+					if (result.isPresent()){
+						this.blockState = this.block.getDefaultState().withProperty(property, result.get());
+					}else{
+						WarpDrive.logger.warn(String.format("Value %s is not allowed for property %s for block %s", propertyValue, propertyKey, state));
+					}
+				}
+				*/
+				// we tried.
+				IProperty<? extends Comparable<?>> property = result.block.getBlockState().getProperty(propertyKey);
+				if (property == null) {
+					WarpDrive.logger.warn(String.format("Found invalid block property %s for block %s", propertyKey, state));
+				} else {
+					boolean found = false;
+					for (int i = 0; i < 16; i++) {    // not efficient, but this would work (and since it's load time, it should not be a problem)
+						IBlockState tmpState = result.block.getStateFromMeta(i);
+						if (tmpState.getProperties().get(property).equals(property.parseValue(propertyValue).orNull())) {
+							result.blockState = tmpState;
+							found = true;
+						}
+					}
+					if (!found) {
+						WarpDrive.logger.warn(String.format("Did not found meta that represent block property %s for block %s", propertyKey, state));
+					}
+				}
+				
+			} else {    //should be digit
+				try {
+					int meta = Integer.parseInt(metaString);
+					result.blockState = result.block.getStateFromMeta(meta);
+				} catch (NumberFormatException e) {
+					WarpDrive.logger.warn(String.format("%s is not a valid number for meta of block %s", metaString, state));
+				}
+			}
+			return result;
+		}
+		
+		public static BlockMatcher fromBlockAndMeta(String blockName, String metaString) {
+			BlockMatcher result = new BlockMatcher();
 
-	@Override
-	public AbstractStructureInstance instantiate(final Random random) {
-		return new SchematicInstance(this, random);
+			result.block = Block.getBlockFromName(blockName);
+			
+			if (result.block == null) {
+				WarpDrive.logger.warn(String.format("Found invalid block in replacement or insertion: %s", blockName));
+				return null;
+			}
+			
+			if(!metaString.equals("")){
+				try {
+					int meta = Integer.parseInt(metaString);
+					result.blockState = result.block.getStateFromMeta(meta);
+				}catch (NumberFormatException e){
+					WarpDrive.logger.warn(String.format("%s is not a valid number for meta of block %s", metaString, blockName));
+				}
+			}
+			return result;
+		}
+		
+		public boolean isMatching(final IBlockState blockStateIn) {
+			if (LOGGING_WORLD_GENERATION) {
+				WarpDrive.logger.info(String.format("matching %s with replacement candidate %s", blockStateIn, this.blockState));
```

Comment:
* This feels only useful for mod developer, consider removing it to reclassing to debug level?
* Assuming we keep that log, consider moving it in the caller?

Reply:
This was a debugging statement that should been removed. It will not be kept due to log spam (total of m*n lines, m=number of replacement + insertion, n=blocks in schematic).